README.md

```diff
@@ -143,69 +264,114 @@ Implementing push notifications requires a few additional snippets of code to en
+method in your app delegate to track when a user opens a push notification. This will handle tracking opens for when the app is backgrounded and the user taps on the notification.
 ```swift
-    func application(application: UIApplication, didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data) {
-        KlaviyoSDK().set(pushToken: deviceToken)
+// be sure to set the UNUserNotificationCenterDelegate to self in the didFinishLaunchingWithOptions method (refer the requesting push notification permission section above for more details on this)
+extension AppDelegate: UNUserNotificationCenterDelegate {
+    // below method will be called when the user interacts with the push notification
+    func userNotificationCenter(
+        _ center: UNUserNotificationCenter,
+        didReceive response: UNNotificationResponse,
+        withCompletionHandler completionHandler: @escaping () -> Void) {
+        // If this notification is Klaviyo's notification we'll handle it
+        // else pass it on to the next push notification service to which it may belong
+        let handled = KlaviyoSDK().handle(notificationResponse: response, withCompletionHandler: completionHandler)
+        if !handled {
+            completionHandler()
+        }
     }
+
+    // below method is called when the app receives push notifications when the app is the foreground
+    func userNotificationCenter(
+        _ center: UNUserNotificationCenter,
+        willPresent notification: UNNotification,
+        withCompletionHandler completionHandler: @escaping (UNNotificationPresentationOptions) -> Void) {
+        if #available(iOS 14.0, *) {
+            completionHandler([.list, .banner])
+        } else {
+            completionHandler([.alert])
+        }
+    }
+}
 ```
+Once your first push notifications are sent and opened, you should start to see _Opened Push_ metrics within your Klaviyo dashboard.
 
-Any users that enable/accept push notifications from your app now will be eligible to receive your custom notifications.
+#### Rich push notifications
 
-To read more about sending push notifications, check out our additional push notification guides.
+>  ℹ️ Rich push notifications are supported in SDK version [2.2.0](https://github.com/klaviyo/klaviyo-swift-sdk/releases/tag/2.2.0) and higher
 
-- [How to set up push notifications](https://help.klaviyo.com/hc/en-us/articles/360023213971)
-- [How to send a push notification campaign](https://help.klaviyo.com/hc/en-us/articles/360006653972)
-- [How to add a push notification to a flow](https://help.klaviyo.com/hc/en-us/articles/12932504108571)
+Rich push notification is the ability to add images to your push notification messages that Apple has supported since iOS 10.
```

Comment:
Reads funny to me a little...
Maybe Rich push notifications are push notifications that include images. Apple has supported this feature since iOS 10.

Reply:
```suggestion
[Rich Push](https://help.klaviyo.com/hc/en-us/articles/16917302437275) is the ability to add images to
push notification messages.
```
is the opening line i've used in the other 2 readmes